crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -850,16 +888,14 @@ impl<'a> Lexer<'a> {
             match c {
                 '$' if self.peek() == Some('{') => {
                     if is_valid_escape_sequence {
-                        self.current.token.value =
-                            TokenValue::String(builder.finish_without_push(self));
+                        self.save_string(true, builder.finish_without_push(self));
                     }
                     self.current.chars.next();
                     return substitute;
                 }
                 '`' => {
                     if is_valid_escape_sequence {
-                        self.current.token.value =
-                            TokenValue::String(builder.finish_without_push(self));
+                        self.save_string(true, builder.finish_without_push(self));
```

Comment:
Note to self: all templates strings are stored because I can't figure out the correct way to handle the `cooked` value for normal template string and tagged template.

tagged`\x` is valid with a cooked value.
`\x` is a syntax error "Bad escape sequence in untagged template literal"